server/interfaces/UserInterface.ts

```diff
@@ -8,14 +8,16 @@ export interface IUserDocument extends Document {
     dialogs: string,
     avatar: string,
     isOnline: boolean,
-    status: string
+    status: string,
+    friends: Array<this | string>
```

Comment:
`this`? Why you use `this` here?

Reply:
self reference, for exzmple ( type User = { friends: Array<string | User> } ) ...... User === this